public/components/freelancer-edit.js

```diff
@@ -535,9 +539,247 @@ function renderCalendar() {
       }
     }
   });
+}
+
+function updateForms(freelancer){
+	freelance = freelancer;
+
+	document.getElementById('email').value = freelancer.email;
+	document.getElementById('description').value = freelancer.description;
+	let temp = "";
+	for( let tag of freelancer.tags){
+		temp+=tag.tagName+",";
+	}
+	document.getElementById('tags').value = temp;
+	document.getElementById('phone').value = freelancer.phone;
+	document.getElementById('address').value = freelancer.address;
+	document.getElementById('pic').value = freelancer.urlPicture;
+	document.getElementById('pricemin').value = freelancer.price.min;
+	document.getElementById('pricemax').value = freelancer.price.max;
+}
 
+class CalendarView extends React.Component {
+	render() {
+		return (
+			<div>
+				<h3>Your emergency schedule:</h3>
+				<ul id="calendar-legend">
+					<li id="calendar-legend-single"> Single dates </li>
+					<li id="calendar-legend-repeated"> Repeated dates </li>
+				</ul>
+				<div id='calendar'></div>
```

Comment:
@Generalbrus there is a problem with the calendar: when first loading the page, no calendar is shown. Only after interacting with the buttons it shows up.